src/emulator/loggingEmulator.ts

```diff
@@ -11,6 +11,20 @@ export interface LoggingEmulatorArgs {
   host?: string;
 }
 
+export interface LogData {
+  user?: any;
+  metadata?: {
+    emulator?: {
+      name?: string;
+    };
+    function?: {
+      name?: string;
+    };
+  };
+}
+
+export type LogDataOrUndefined = LogData | undefined;
```

Comment:
IMO we don't need this type, it can be cleanly expressed as `LogData | undefined` where it's used.

Reply:
I was doing that as first but we ended up with `LogData | undefined = undefined` in like 1000 places, which just felt weird. I could go either way.